app/src/main/java/ch/epfl/sweng/SDP/matchmaking/Matchmaker.java

```diff
@@ -1,73 +1,142 @@
      */
-    Matchmaker() {
-        myRef = Database.INSTANCE.getReference("rooms");
+    private Matchmaker(ConstantsWrapper constantsWrapper) {
+
+        this.constantsWrapper = constantsWrapper;
+        this.reference = constantsWrapper.getReference("rooms");
 
         // Read from the database
-        myRef.addListenerForSingleValueEvent(new ValueEventListener() {
+        reference.addListenerForSingleValueEvent(new ValueEventListener() {
 
             @Override
-            public void onDataChange(@NonNull DataSnapshot dataSnapshot) {
+            public void onDataChange(DataSnapshot dataSnapshot) {
                 // This method is called once with the initial value and again
-                // whenever data at this location is updated.
-                HashMap<String, Object> map = (HashMap<String, Object>) dataSnapshot.getValue();
+                // whenever data at this location is updated;
                 // Update room
 
-                Log.d("1", map.toString());
             }
 
             @Override
-            public void onCancelled(@NonNull DatabaseError error) {
+            public void onCancelled(DatabaseError error) {
                 // Failed to read value
             }
         });
+
+
     }
 
     /**
-     * Join a room.
-     *
-     * @param roomId the id of the room.
+     * join a room.
      */
-    public void joinRoom(String roomId) {
-        // FirebaseUser currentFirebaseUser = FirebaseAuth.getInstance().getCurrentUser() ;
-        DatabaseReferenceBuilder builder = new DatabaseReferenceBuilder(myRef);
-        builder.addChildren(roomId + ".users.123").build().setValue("InRoom");
+    public Boolean joinRoom() {
+
+            Boolean successful = false;
+        HttpURLConnection connection = null;
+
+        try {
+                //Create connection
+                String urlParameters = "userId=" + URLEncoder.encode(constantsWrapper.getFirebaseUserId(), "UTF-8");
+                URL url = new URL("https://us-central1-gyrodraw.cloudfunctions.net/joinGame?" + urlParameters);
+                connection = createConnection(url);
+
+                //Send request
+                DataOutputStream wr = new DataOutputStream(
+                        connection.getOutputStream());
+                wr.close();
+
+                //Get Response
+                InputStream is = connection.getInputStream();
+                BufferedReader rd = new BufferedReader(new InputStreamReader(is));
+                StringBuilder response = new StringBuilder(); // or StringBuffer if Java version 5+
+                String line;
+
+                if (connection.getResponseCode() == HttpURLConnection.HTTP_OK) {
+                    // OK
+                    successful = true;
+                    // otherwise, if any other status code is returned, or no status
+                    // code is returned, do stuff in the else block
+                }
+
+                while ((line = rd.readLine()) != null) {
```

Comment:
Do you need these lines? You never do anything with response.

Reply:
You are right